docs/webui_overriding.md

```diff
@@ -0,0 +1,139 @@
+a Lit HTML template we need a way to directly modify the source file. This is
+done with a Lit mangler.
+
+Create a file in `chromium_src` for `path/to/your/file.html.ts` at `chromium_src/path/to/your/file.html.ts`
+
+```ts
+import mangle from 'lit-mangler'
+
+mangle(element => {
+  element.textContent = element.textContent.replaceAll("Chrome", "Brave")
+}, literal => literal.text.includes("id='thing-i-want-to-edit'"))
+```
+
+The `mangle` function extracts all templates from the `.html.ts` file and loads
+them into an HTMLElement, so you can manipulate them using the DOM APIs.
+
+The second argument to this function is optional but lets you select what
+template you want to edit (in the case of multiple nested templates).
+
+For example:
+```ts
+// this template literal from upstream:
+html`<div class="container">
+  ${this.children.map(c => html`<li class="chrome-item">${c.text}</li>`)}
+</div>`
+
+// would have two templates
+// 1. For the container
+// 2. For the children, iterated over the array.
+
+
+// in our mangler:
+// We'll make the following changes with a Lit Mangler:
+// 1. Add a `.brave` class to the `.container` element
+// 2. Add an `id` attribute to the `li` elements
+// 3. Wrap the child text in a span
+
+import mangle from 'lit-mangler'
+
+mangle(e => {
+  const container = e.querySelector('.container')
+
+  // Its good practise to throw an error when the mangler can't find something
+  // - it will cause the build to fail, so we'll know quickly that something
+  // went wrong.
+  if (!container) throw new Error("Couldn't find the container")
+// Note: We use this predicate to select the right template.
+}, t => t.text.includes('class="container"'))
+
+mangle(e => {
+  // Note: The element passed to the mangler is a document fragment, as there
+  // could be multiple root elements in a template.
+  const li = e.querySelector('li')
+
+  // Assuming our children have an id element. Note: We set this to a string
+  // literal, because we're editing the source text of the .html.ts file.
+  li.setAttribute('id', '${c.id}');
+
+  // Similarly, we need to escape the template interpolation here, because we
+  // want it to happen at runtime, not now!
+  // If we mess something up we'll get a typescript error (as the mangled source
+  // file still runs through the type checker)!
+  li.innerHTML = `<span>
+    \${c.text}
+  </span>`
+// select the list items
+}, t => t.text.startsWith('<li'))
+```
+
+## Polymer Template Modifications
+
+Polymer templates are just HTML Template elements, so you can easily modify them
+with a `chromium_src` override (for the JS file) or more easily via the
+`polymer_overriding` utils in `//brave/ui/webui/resources/polymer_overriding.ts`
+
+## Adding New Files
+
+Often, you won't only be changing upstream elements but will also need to add
+new elements to the page to get things looking right.
+
+In this scenario, its best to add a new build target to Brave and patch it into
```

Comment:
Consider adding a part discouraging putting all brave logic in a chromium_src override? Something like:
"Ideally `chromium_src` WebUI files should contain as little code as possible: just enough to modify chromium components and load brave components from the regular brave source tree (e.g. `brave/browser/resources/`). This is the same guideline as for chromium_src C++ files. This facilitates readability as it helps avoid having to jump from `brave/chromium_src/chrome/browser/resources/` to `brave/browser/resources/` to follow brave-specific WebUI code."